master/buildbot/reporters/telegram.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+from future.utils import string_types
+
+from twisted.internet import defer
+from twisted.python import log
+
+from buildbot import config
+from buildbot.reporters.utils import getDetailsForBuild
+from buildbot.process.results import statusToString
+from buildbot.util.service import BuildbotService
+
+from telegram import Bot
+
+class TelegramStatus(BuildbotService):
+    name = "Telegram"
+
+    def checkConfig(self, auth_token, chat_id, **kwargs):
+        if not isinstance(auth_token, string_types):
+            config.error('auth_token must be a string')
+
+    @defer.inlineCallbacks
+    def reconfigService(self, auth_token, chat_id, **kwargs):
+        yield BuildbotService.reconfigService(self, **kwargs)
+
+        self.auth_token = auth_token
```

Comment:
are all these data member part of public API? if not, please prepend them with `_`, like `self._auth_token`